src/bitmessagemain.py

```diff
@@ -1,49 +1,55 @@
 #!/usr/bin/python2.7
-# Copyright (c) 2012-2016 Jonathan Warren
-# Copyright (c) 2012-2018 The Bitmessage developers
-# Distributed under the MIT/X11 software license. See the accompanying
-# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+# pylint: disable=no-self-use,too-many-branches,too-many-statements,too-many-locals
+"""
+bitmessagemain.py
+=================
 
-# Right now, PyBitmessage only support connecting to stream 1. It doesn't
-# yet contain logic to expand into further streams.
+Copyright (c) 2012-2016 Jonathan Warren
+Copyright (c) 2012-2018 The Bitmessage developers
+Distributed under the MIT/X11 software license. See the accompanying
+file COPYING or http://www.opensource.org/licenses/mit-license.php.
 
-# The software version variable is now held in shared.py
+Right now, PyBitmessage only support connecting to stream 1. It doesn't
+yet contain logic to expand into further streams.
 
+The software version variable is now held in shared.py
+
+"""
+
+from __future__ import absolute_import
+
+import ctypes
+import errno
+import getopt
 import os
+import signal
+import socket
 import sys
+import threading
+from random import randint
```

Comment:
please give a space between import and from statements.

Reply:
Personally, I'm not in favour of such a formatting option. Neither is isort by default. It may be possible to get isort to do this but we should invite more general discussion about non-violating import formatting issues and update the isort config accordingly.

Reply:
No need for space. PEP8 requires spaces between groups, e.g. system, libs, internal imports, not `import ..` and `from .. import`.